test/e2e/resource_validation_test.go

```diff
@@ -0,0 +1,226 @@
+		var name = "abcdef-123456789-123456789-123456789-123456789-123456789-123456789-123456789"
+		// Create the API.
+		memberClusterName := &clusterv1beta1.MemberCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: name,
+			},
+			Spec: clusterv1beta1.MemberClusterSpec{
+				Identity: rbacv1.Subject{
+					Name:      "fleet-member-agent-cluster-1",
+					Kind:      "ServiceAccount",
+					Namespace: "fleet-system",
+					APIGroup:  "",
+				},
+			},
+		}
+		By(fmt.Sprintf("expecting denial of CREATE API %s", name))
+		err := hubClient.Create(ctx, memberClusterName)
+		var statusErr *k8serrors.StatusError
+		Expect(errors.As(err, &statusErr)).To(BeTrue(), fmt.Sprintf("Create API call produced error %s. Error type wanted is %s.", reflect.TypeOf(err), reflect.TypeOf(&k8serrors.StatusError{})))
+		Expect(statusErr.Status().Message).Should(ContainSubstring("metadata.name max length is 63"))
+	})
+
+	It("should allow creating API with valid name size", func() {
+		var name = "abc-123456789-123456789-123456789-123456789-123456789-123456789"
+		// Create the API.
+		memberClusterName := &clusterv1beta1.MemberCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: name,
+			},
+			Spec: clusterv1beta1.MemberClusterSpec{
+				Identity: rbacv1.Subject{
+					Name:      "fleet-member-agent-cluster-1",
+					Kind:      "ServiceAccount",
+					Namespace: "fleet-system",
+					APIGroup:  "",
+				},
+			},
+		}
+		Expect(impersonateHubClient.Create(ctx, memberClusterName)).Should(Succeed())
+		Expect(impersonateHubClient.Get(ctx, types.NamespacedName{Name: memberClusterName.Name}, memberClusterName)).Should(Succeed())
+		Expect(impersonateHubClient.Delete(ctx, memberClusterName)).Should(Succeed())
+		ensureMemberClusterAndRelatedResourcesDeletion(name)
+	})
+
+	It("should deny creating API with invalid name starting with non-alphanumeric character", func() {
+		var name = "-abcdef-123456789-123456789-123456789-123456789-123456789"
+		// Create the API.
+		memberClusterName := &clusterv1beta1.MemberCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: name,
+			},
+			Spec: clusterv1beta1.MemberClusterSpec{
+				Identity: rbacv1.Subject{
+					Name:      "fleet-member-agent-cluster-1",
+					Kind:      "ServiceAccount",
+					Namespace: "fleet-system",
+					APIGroup:  "",
+				},
+			},
+		}
+		By(fmt.Sprintf("expecting denial of CREATE API %s", name))
+		err := hubClient.Create(ctx, memberClusterName)
+		var statusErr *k8serrors.StatusError
+		Expect(errors.As(err, &statusErr)).To(BeTrue(), fmt.Sprintf("Create API call produced error %s. Error type wanted is %s.", reflect.TypeOf(err), reflect.TypeOf(&k8serrors.StatusError{})))
+		Expect(statusErr.Status().Message).Should(ContainSubstring("a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')"))
```

Comment:
no need to compare the full error strings and it's not reliable cause the package may change the error message.

Checking if have "a lowercase RFC 1123 subdomain" should be fine.